src/main/scripts/src/app/shared/permissions/table/permissions-table.component.html

```diff
@@ -7,27 +7,33 @@
     </tr>
   </thead>
   <tbody>
-    <tr *ngFor="let entity of entities">
+    <tr *ngFor="let entity of entities; let i = index" [attr.data-index]="i">
       <td>{{entity.name}}</td>
-      <td *ngFor="let role of roles" class="role">
-        <input type="checkbox" title="{{role}}" [attr.checked]="entity.roles.indexOf(role) > -1 ? '': null"/>
+      <td *ngFor="let role of roles"  class="role">
+        <div class="material-switch pull-left">
+          <input id="ch{{entity.name+role}}" type="checkbox" title="{{role}}" [attr.checked]="entity.roles.indexOf(role) > -1 ? '': null" (change)="changePermission(entity.name, role, $event)"/>
+          <label for="ch{{entity.name+role}}" class="label-primary"></label>
+        </div>
       </td>
       <td>
-        <button (click)="onRemove(entity.name)" class="btn btn-remove">Remove</button>
+        <button (click)="onRemove(i)" class="btn btn-danger">Remove</button>
```

Comment:
The `entities` variable keeps the array of entity objects. When I want to remove an item of an array I need the index. This way I do not have to get the index of the element I want to remove in the `onRemove` method.
If there is only the name of an entity in `onRemove` method (as it was before) I need to iterate the whole `entities` array and look for the entity object with the given name to get the index.